lib/presentation/features/home/home_screen.dart

```diff
@@ -19,20 +18,27 @@ class HomeScreen extends StatelessWidget {
         title: const Text("MarvellApp"),
       ),
       body: BlocProvider(
-        create: (_) =>
-            HomeBloc(getCharactersUseCase: GetCharactersUseCase(repository: DioMarvellRepository()))..add(GetHeroes()),
+        create: (_) => HomeBloc(getCharactersUseCase: GetCharactersUseCase(repository: DioMarvellRepository()))
+          ..add(ReadyForData()),
         child: BlocBuilder<HomeBloc, HomeState>(
           builder: (context, state) {
             Widget child = const Center(
               child: CircularProgressIndicator(),
             );
-            if (state.loading) return child;
-            if (state.error == null && !state.loading) {
-              child = ListCharacters(list: state.characters);
-            } else {
-              child = const CustomErrorWidget();
+            if (state.characters != null && state.error == null) {
+              return ListCharacters(list: state.characters, error: state.error, hasReachedMax: state.hasReachedMax);
+            }
+            if (state.characters == null && state.error == null) {
```

Comment:
These conditions are still looks not so clear.
Showing the loading based on `state.characters == null` instead of the checking `state.loading` is a bit weird. 
Is that possible to write it similar to this? 
```
if (state.loading) return CircularProgressIndicator();
if (state.error != null) return ErrorWidget();
...
return ListCharacters(...);
```

Reply:
done